.github/workflows/build-and-test.yml

```diff
@@ -1,23 +1,49 @@
 name: Build and Tests
 
-on: [push, pull_request]
+on:
+  push:
+    branches: 
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
 
 jobs:
-  build:
+  test:
     runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        test-scenario: [short, long]
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
       - name: install-go
         run: sudo snap install go --classic
+
       - name: build
         run: go build ./...
+
       - name: apt-update
         run: sudo apt update
+
       - name: install-test-dependencies
         run: sudo apt install -y snapd germinate mtools debootstrap fdisk gdisk qemu-system-aarch64 qemu-user-static ubuntu-dev-tools
-      - name: test
-        run: sudo go test -timeout 0 -v -race -coverprofile=coverage.out -covermode=atomic ./...
+
+      - name: short tests
+        if: ${{ matrix.test-scenario == 'short' }}
+        run: sudo go test -timeout 0 -v -test.short -coverprofile=.coverage/coverage-short.out -covermode=atomic ./...
```

Comment:
How does this work? Since I see some conditionals for when `testing.Short()` is on. But does this mean that the long tests execute all the long AND short tests? Or is there a way to just get the long ones run?

Reply:
Yes, the "long tests" are also executing the short ones. Unfortunately there is no practical way to exclude short ones. But  since short ones takes less than 10s this is negligible comparing to the long ones. Moreover, since I upload the coverage in both cases, this is nice that long tests also cover the short ones so the final coverage is the complete one (and not only the coverage of long tests).